be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -16,12 +16,55 @@
 
 #include "fs/fs_util.h"
 #include "storage/lake/filenames.h"
+#include "storage/lake/meta_file.h"
 #include "storage/lake/persistent_index_memtable.h"
 #include "storage/lake/persistent_index_sstable.h"
 #include "storage/lake/tablet_manager.h"
+#include "storage/sstable/iterator.h"
+#include "storage/sstable/merger.h"
+#include "storage/sstable/options.h"
+#include "storage/sstable/table_builder.h"
 
 namespace starrocks::lake {
 
+Status KeyValueMerger::merge(const std::string& key, const std::string& value) {
+    IndexValueWithVerPB index_value_ver;
+    if (!index_value_ver.ParseFromString(value)) {
+        return Status::InternalError("Failed to parse index value ver");
+    }
+
+    auto version = index_value_ver.versions(0);
+    auto index_value = index_value_ver.values(0);
+    if (_key == key) {
+        if (_index_value_vers.empty()) {
+            _index_value_vers.emplace_front(version, index_value);
+        } else if (version > _index_value_vers.front().first) {
+            std::list<std::pair<int64_t, IndexValue>> t;
+            t.emplace_front(version, index_value);
+            _index_value_vers.swap(t);
+        }
+    } else {
+        build_index_value_vers();
+        _key = key;
+        _index_value_vers.emplace_front(version, index_value);
+    }
+    return Status::OK();
+}
+
+void KeyValueMerger::build_index_value_vers() {
```

Comment:
may give a better name for this method. It doesn't look like build the `_index_value_vers` member variable, but actually flushing the `_index_value_vers` into builder.